### PR TITLE
add verification of lengths

### PR DESCRIPTION
**Pull request by juliangruber**
cc @mafintosh 

I did this primarily so dat.haus can use this module to determine if a link is valid

### CODE REVIEW

**Comment by juliangruber** [comment]
I thought about returning `null` instead of throwing, but I can see that creating even weirder errors when you don't account for it

File: index.js [hunk 1]
```diff
@@ -2,17 +2,20 @@
 
 var baseX = require('base-x')
 var Buffer = require('safe-buffer').Buffer
+var assert = require('assert')
 
 var alphabet = '0123456789abcdefghijklmnopqrstuvwxyz'
 var encoding = baseX(alphabet)
 
 exports.encode = function (buf) {
+  assert(buf.length === 32, 'Invalid buffer')
```
Comment (mafintosh) [inline index.js]: how big is assert when browserified?

File: index.js [hunk 2]
```diff
@@ -2,17 +2,20 @@
...
   buf = Buffer.from(buf)
   return encoding.encode(buf)
 }
 
 exports.decode = function (str) {
   str = str.slice(str.lastIndexOf('/') + 1)
   if (str.length === 64) return Buffer.from(str, 'hex')
```
Comment (mafintosh) [inline index.js]: maybe regex that this is hex as well? `/[0-9a-f]{64}/i.test(str)`

**Comment by mafintosh** [comment]
I think throwing is the way to go for this.

**Comment by juliangruber** [comment]
i removed assert because it's in total ~30kb bundled, also it wasn't adding much here

**Comment by juliangruber** [comment]
i improved the hex check based on your suggestion as well

**Comment by juliangruber** [comment]
thanks @mafintosh! published as 2.0.0

**Pull request merged**